obsreport/obsreport_receiver.go

```diff
@@ -237,10 +237,12 @@ func ReceiverContext(
 	}
 
 	if useNew {
-		mutators := []tag.Mutator{
-			tag.Upsert(tagKeyReceiver, receiver, tag.WithTTL(tag.TTLNoPropagation)),
-			tag.Upsert(tagKeyTransport, transport, tag.WithTTL(tag.TTLNoPropagation)),
+		mutators := make([]tag.Mutator, 0, 2)
+		mutators = append(mutators, tag.Upsert(tagKeyReceiver, receiver, tag.WithTTL(tag.TTLNoPropagation)))
+		if transport != "" {
```

Comment:
Not totally sure about using the empty string to signify an optional parameter, but it seems fairly idiomatic from what I can tell, and has the advantage here of being a non-breaking change.